omod-common/src/main/java/org/openmrs/module/webservices/rest/web/api/RestService.java

```diff
@@ -27,17 +27,33 @@
 	/**
 	 * Parses a representation requested by the client via the http request
 	 * 
-	 * @param requested
+	 * @param requested the requested representation
 	 * @return
 	 * <strong>Should</strong> get ref representation when specified
 	 * <strong>Should</strong> get default representation when specified
 	 * <strong>Should</strong> get full representation when specified
 	 * <strong>Should</strong> get a named representation when specified
 	 */
-	public Representation getRepresentation(String requested);
+	Representation getRepresentation(String requested);
 	
+	/**
+	 * Gets resource by name.
```

Comment:
Do we really need to make any changes to this class?

Reply:
Because both Name and UUID are strings we could get away with the `getResourceByName(String name)` and not make any changes to the class but I considered the readability of the code. if we are expecting a name or UUID it is better we have a method that does just that getResourceByNameOrUuid because the Rest Controller needs to call exactly one method.

Reply:
How about doing general improvements in a ticket and pull request of their own?

Reply:
Alright, I will do that.